src/Microsoft.Identity.Web/ClaimsPrincipalFactory.cs

```diff
@@ -107,5 +107,36 @@ public static ClaimsPrincipal FromTenantIdAndObjectId(string tenantId, string ob
                     }));
             }
         }
+
+        /// <summary>
+        /// Instantiate a <see cref="ClaimsPrincipal"/> from a username and password.
+        /// This can be used for ROPC flow for testing purposes.
+        /// </summary>
+        /// <param name="username">UPN of the user for example username@domain.</param>
+        /// <param name="password">Password for the user.</param>
+        /// <returns>A <see cref="ClaimsPrincipal"/> containing these two claims.</returns>
+        public static ClaimsPrincipal FromUsernamePassword(string username, string password)
+        {
+            if (AppContextSwitches.UseClaimsIdentityType)
+            {
+#pragma warning disable RS0030 // Do not use banned APIs
+                return new ClaimsPrincipal(
+                    new ClaimsIdentity(new[]

```

Comment:
You should instantiate a `CaseSensitiveClaimsIdentity`, not a `ClaimsIdentity`.
Please don't disable the warning. just do what it tells you to do.